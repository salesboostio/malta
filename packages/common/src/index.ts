export {
  getAPI,
  getAuthURL,
  getCatalogURL,
  getEnv,
  getPixelURL,
  isDevelopment,
  isLocalhost,
  isProduction,
  isStaging,
} from "./env";
export {
  SalesboostRouter,
  useLink,
  createQueryString,
} from "./router";
