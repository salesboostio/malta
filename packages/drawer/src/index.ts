export { ServiceDrawer } from "./ServiceDrawer";
export { SettingDrawer } from "./SettingDrawer";
