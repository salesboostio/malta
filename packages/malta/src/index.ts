export {
  App,
  Block,
  Box,
  Br,
  Flex,
  Grid,
  Icon,
  Image,
  Split,
  Text,
} from "./core";

export {
  theme,
  Button,
  Card,
  Checkbox,
  Drawer,
  Header,
  IconButton,
  Input,
  Message,
  Portal,
  RadioButton,
  Select,
  Spinner,
  Switch,
  message,
  notify,
  dialog,
} from "./salesboost";
