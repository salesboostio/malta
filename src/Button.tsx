import * as React from "react";
import styled from "styled-components";
import {
  colors,
  fontSizes
} from "./theme";
const {
  borderColor,
  borderRadius,
  borders,
  color,
  fontSize,
  height,
  space,
  width,
} = require("styled-system");
import { generateQuery } from "./tools/mediaQuery";

const StyledButton = styled.button`
  align-items: center;
  cursor: pointer;
  display: flex;
  justify-content: center;
  ${borderColor}
  ${borderRadius}
  ${borders}
  ${color}
  ${fontSize}
  ${space}
  ${width}
  ${height}

  ${generateQuery("max-width", "sm", `
    font-size: ${fontSizes.h5}px;
    height: 52px;
  `)};

  &:hover {
    background-color: ${(props: IProps) => props.hoverColor};
  }

  &:focus {
    background-color: ${(props: IProps) => props.focusColor};
  }
`;

export interface IProps {
  onClick?: React.MouseEventHandler<HTMLElement>;
  type?: string;
  // borderColor
  borderColor?: string;
  // borderRadius
  borderRadius?: number | string;
  // borders
  border?: string;
  borderTop?: string;
  borderRight?: string;
  borderBottom?: string;
  borderLeft?: string;
  // color
  color?: string;
  bg?: string;
  // fontSize
  fontSize?: Array<number | string> | number | string;
  // height
  height?: Array<number | string> | number | string;
  // space
  m?: Array<number | string> | number | string;
  mt?: Array<number | string> | number | string;
  mr?: Array<number | string> | number | string;
  mb?: Array<number | string> | number | string;
  ml?: Array<number | string> | number | string;
  mx?: Array<number | string> | number | string;
  my?: Array<number | string> | number | string;
  p?: Array<number | string> | number | string;
  pt?: Array<number | string> | number | string;
  pr?: Array<number | string> | number | string;
  pb?: Array<number | string> | number | string;
  pl?: Array<number | string> | number | string;
  px?: Array<number | string> | number | string;
  py?: Array<number | string> | number | string;
  // width
  width?: Array<number | string> | number | string;
  // options
  hoverColor?: string;
  focusColor?: string;
}

export class Button extends React.Component<IProps, any> {
  public static defaultProps: IProps = {
    onClick: () => { },
    // borderColor
    borderColor: "transparent",
    // borderRadius
    borderRadius: "30px",
    // borders
    border: "none",
    // color
    color: "white.standard",
    bg: "blue.royal",
    // fontSize
    fontSize: "h3",
    // height
    height: "60px",
    // space
    m: 0,
    mb: 2,
    px: 3,
    // width
    width: "auto",
    // options
    hoverColor: colors.blue.deep,
    focusColor: colors.blue.dark,
  };

  public render() {
    return (
      <StyledButton
        {...this.props}
      />
    );
  }
}
