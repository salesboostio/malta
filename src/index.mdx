---
name: Introduction
route: /
---

# Malta [[Github]](https://github.com/salesboostio/malta)

Salesboost UI library. Atomic Design 패러다임을 적용해 컴포넌트들이 추상화의
수준에 맞춰 카테고리가 나뉘어져 있다. 카테고리의 hierarchy는 `atom` - `molecule` - `organism` - `template`
순이며 모든 컴포넌트는 자신보다 상위레벨의 컴포넌트를 참조해선 안된다.

- [Atom](http://localhost:3000/src-atoms-index): Text, Box, Image...
- [Molecules](http://localhost:3000/src-molecules-index): Shadow, RoundedBox, Label, Button, Span, HyperLink, Chip, ListContainer, ListItem...
- [Organisms](http://localhost:3000/src-organisms-index): Card, FixedHeader, FeedItem, GridItem, Slider...
- [Templates](http://localhost:3000/src-templates-index): SalesBoostFooter, SalesBoostHeader, CatalogNav...



- [Misc](http://localhost:3000/src-misc-index): 아직 분류되지 않은 draft 컴포넌트들, deprecate 컴포넌트들을 모아놓은 폴더
- [Commons](http://localhost:3000/src-commons-index): 공통으로 사용하는 App, Theme, Other utils...

## WorkFlow

## Shell Command
```
~/malta$ tsc             # 컴파일
~/malta$ ./bin/todo      # /src내에 존재하는 todo를 grep해서 src/index.mdx에 업데이트
```

## Docz [[link]](https://www.docz.site/)
프로젝트 다큐멘테이션 라이브러리며 mdx를 사용한다. 소스 폴더내에 어디든 .mdx 파일을 생성하기만 하면
자동으로 docz 웹페이지로 변환시켜준다.
```
npm run docz
npm run docz build
```


## References
