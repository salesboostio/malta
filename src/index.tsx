export { Absolute } from "./misc/Absolute";
export { App } from "./commons/App";
export { Box } from "./atoms/Box";
export { Chip } from "./molecules/Chip";
export { Container } from "./atoms/Container";
export { Flex } from "./atoms/Flex";
export { FullScreen } from "./misc/FullScreen";
export { Input } from "./atoms/Input";
export { Hide } from "./misc/Hide";
export { HyperLink } from "./molecules/HyperLink";
export { Image } from "./atoms/Image";
export { Overlay } from "./misc/Overlay";
export { Relative } from "./misc/Relative";
export { Rounded } from "./misc/Rounded";
export { SalesboostFooter } from "./templates/SalesboostFooter";
export { Shadow } from "./misc/Shadow";
export { Span } from "./molecules/Span";
export { Text } from "./atoms/Text";