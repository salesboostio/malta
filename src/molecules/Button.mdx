---
name: Button
menu: Molecules
---

import { Flex, App, Chip, Button } from "../../dist"
import { Playground } from "docz"

# Button

버튼입니다. onClick Prop에 콜백함수를 넣어주세요.

## Dependencies
<App>
<Flex>
<Chip bg="blue.deep" color="#FFF">Box</Chip>
</Flex>
</App>

## Properties

|Property|Type|Default|
|-|-|-|
| color | string | white.standard |
| height | ResponsiveValues | [52px, 60px] |
| width | ResponsiveValue | fit-content |
| fontSize | ResponsiveValues | [h5, h3] |
| hoverColor | String | color.blue.deep |
| focusColor | String | color.blue.dark |
| bg | String | blue.royal |
| onClick | function | ()=>{} |

- hoverColor, focusColor는 theme적용이 안되므로 theme을 import해서 color.*으로 사용했습니다.

## Usecase

<Playground>
  <App>
    <Button>Click me</Button>
  </App>
</Playground>

## Best Practices

## Test
