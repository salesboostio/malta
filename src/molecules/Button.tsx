import * as React from "react";
import styled from "styled-components";
import { border, BorderProps, borderColor, BorderColorProps, borderRadius, BorderRadiusProps, fontSize, FontSizeProps } from "styled-system";
import { Box } from "../atoms/Box";
import { colors } from "../commons/theme";

interface HoverColorProps {
  hoverColor?: string;
}
interface FocusColorProps {
  focusColor?: string;
}

type Props = BorderColorProps & BorderRadiusProps & BorderProps & FontSizeProps & HoverColorProps & FocusColorProps;

export const Button = styled(Box)<Props>`
  display: flex;
  cursor: pointer;
  align-items: center;
  justify-content: center;
  font-weight: 500;
  width: fit-content;
  &:hover {
    background: ${(props: Props) => props.hoverColor};
  }
  &:focus {
    background: ${(props: Props) => props.focusColor};
  }
  ${fontSize}
  ${borderColor}
  ${borderRadius}
  ${border}
`;

Button.defaultProps = {
  onClick: () => { },
  borderColor: "transparent",
  borderRadius: ["26px", "30px"],
  border: "none",
  color: "white.standard",
  bg: "blue.royal",
  fontSize: ["h5", "h3"],
  height: ["52px", "60px"],
  m: 0,
  mb: 2,
  px: 3,
  hoverColor: colors.blue.deep,
  focusColor: colors.blue.dark
};
