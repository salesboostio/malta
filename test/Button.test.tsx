
import * as React from "react";
import { create } from "react-test-renderer";
import { Button } from "../src/Button";

test("Button", () => {
    const node = create(<Button>Hello World!</Button>).toJSON();
    expect(node).toMatchSnapshot();
});
