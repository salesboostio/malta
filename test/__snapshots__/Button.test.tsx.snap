// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<button
  className="sb-button sc-bdVaJa kKJYgi"
>
  Hello World!
</button>
`;
